util/br/pkg/restore/restore.go

```diff
@@ -0,0 +1,373 @@
+			r.log.Error("restore failed", zap.String("error code", resp.GetCode().String()))
+			return restoreFailed
+		}
+
+		if resp.GetCode() == meta.ErrorCode_SUCCEEDED {
+			return nil
+		}
+
+		leader := resp.GetLeader()
+		if leader == meta.ExecResp_Leader_DEFAULT {
+			return LeaderNotFoundError
+		}
+
+		// we need reconnect the new leader
+		err = r.client.Open(hostaddrToString(leader))
+		if err != nil {
+			return err
+		}
+		count--
+	}
+}
+
+func (r *Restore) cleanupOriginal() error {
+	g, _ := errgroup.WithContext(context.Background())
+	for _, addr := range r.config.StorageAddrs {
+		cmd := r.backend.RestoreStoragePreCommand(r.config.StorageDataDir + "/nebula")
+		ipAddr := strings.Split(addr, ":")[0]
+		g.Go(func() error { return ssh.ExecCommandBySSH(ipAddr, r.config.StorageUser, cmd, r.log) })
+	}
+
+	for _, addr := range r.config.MetaAddrs {
+		cmd := r.backend.RestoreMetaPreCommand(r.config.MetaDataDir + "/nebula")
+		ipAddr := strings.Split(addr, ":")[0]
+		g.Go(func() error { return ssh.ExecCommandBySSH(ipAddr, r.config.StorageUser, cmd, r.log) })
+	}
+
+	err := g.Wait()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *Restore) stopCluster() error {
+	g, _ := errgroup.WithContext(context.Background())
+	for _, addr := range r.config.StorageAddrs {
+		cmd := "cd " + r.config.StorageRootDir + " && scripts/nebula.service stop storaged"
+		ipAddr := strings.Split(addr, ":")[0]
+		g.Go(func() error { return ssh.ExecCommandBySSH(ipAddr, r.config.StorageUser, cmd, r.log) })
+	}
+
+	for _, addr := range r.config.MetaAddrs {
+		cmd := "cd " + r.config.MetaRootDir + " && scripts/nebula.service stop metad"
+		ipAddr := strings.Split(addr, ":")[0]
+		g.Go(func() error { return ssh.ExecCommandBySSH(ipAddr, r.config.MetaUser, cmd, r.log) })
+	}
+
+	err := g.Wait()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *Restore) startMetaService() error {
+	g, _ := errgroup.WithContext(context.Background())
+	for _, addr := range r.config.MetaAddrs {
+		cmd := "cd " + r.config.MetaRootDir + " && scripts/nebula.service start metad &>/dev/null &"
+		ipAddr := strings.Split(addr, ":")[0]
+		g.Go(func() error { return ssh.ExecCommandBySSH(ipAddr, r.config.MetaUser, cmd, r.log) })
+	}
+
+	err := g.Wait()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *Restore) startStorageService() error {
+	g, _ := errgroup.WithContext(context.Background())
+	for _, addr := range r.config.StorageAddrs {
+		cmd := "cd " + r.config.StorageRootDir + " && scripts/nebula.service start storaged &>/dev/null &"
+		ipAddr := strings.Split(addr, ":")[0]
+		g.Go(func() error { return ssh.ExecCommandBySSH(ipAddr, r.config.StorageUser, cmd, r.log) })
+	}
+
+	err := g.Wait()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *Restore) RestoreCluster() error {
+
+	err := r.downloadMetaFile()
+	if err != nil {
+		r.log.Error("download meta file failed", zap.Error(err))
+		return err
+	}
+
+	m, err := r.restoreMetaFile()
```

Comment:
What is in the meta file for now?

Reply:
You can see the contents of the corresponding meta file in the [PR of nebula-common](https://github.com/vesoft-inc/nebula-common/pull/288).